Разработка консольного приложения

damirs11 opened the pull request.

Commit:
done

Commit:
start working on JAXB class

damirs11 committed:
Add PersonsList_class for multiple Person_model serialization

damirs11 committed:
Add PersonsList_class for multiple Person_model serialization

dreambot92 commented on src/main/java/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,91 @@
+import com.company.Models.Documents.Outgoing;
+import com.company.Models.Documents.Task;
+
+import java.util.*;
```
Надо бы настроить идею.. так ты подгружаешь все пакеты с java.util.*
см. 3.3.1 в code style

damirs11 replied:
Исправил

damirs11 replied:
Исправил

dreambot92 commented on src/main/java/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,91 @@
+
+public class DocumentFactory  {
+
+    public DocumentFactory() {
```
Вообще нам нужен единственный экземпляр фабрики. Конструктор не нужен
Сделай setOfDocuments статичным

dreambot92 replied:
как и все методы в классе

azamat2 replied:
а разве фабрика вообще должна хранить в себе документы? по-моему вся ее задача - предоставлять статичные методы, конструирующие объект

dreambot92 replied:
+++

damirs11 replied:
Исправил

dreambot92 commented on src/main/java/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,91 @@
+
+        Document doc = null;
+
+        if(Objects.equals(docType.getName(), Outgoing.class.getName() ))
```
code style 4.1

damirs11 replied:
Исправил

dreambot92 commented on src/main/java/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,91 @@
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class DocumentFactory  {
```
javadoc-и бы

dreambot92 commented on src/main/java/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,91 @@
+    }
+    public Document generateData(Document doc) throws DocumentExistsException {
+
+        String[] names = {
```
Сейчас массив имен создается каждый раз при вызове метода. Это ни к чему.. вынеси в константы класса

damirs11 replied:
Исправил

dreambot92 commented on src/main/java/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,91 @@
+        };
+
+        GregorianCalendar gc = new GregorianCalendar();
+        int year = 2010 + (int) Math.round(Math.random() * (2020 - 2010));
```
Магические цифры в константы

dreambot92 replied:
Установить SonarLint в идею будет не лишним

damirs11 replied:
Поставил пока что на своем ПК 
Удобная штука, спасибо

dreambot92 commented on src/main/java/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,91 @@
+        int year = 2010 + (int) Math.round(Math.random() * (2020 - 2010));
+        int dayOfYear = 1 + (int) Math.round(Math.random() * (gc.getActualMaximum(Calendar.DAY_OF_YEAR) - 1));
+        gc.set(Calendar.YEAR, year);
+        gc.set(Calendar.DAY_OF_YEAR, dayOfYear);
```
Я бы логику по получению случайной даты вынес в отдельный метод

damirs11 replied:
Исправил

dreambot92 commented on src/main/java/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,91 @@
+        doc.setAuthor( names[(int) (Math.random() * (names.length - 1))] );
+        doc.setRegDate(gc.getTime());
+
+        doc.setReg_idDoc((int) (Math.random() * 1000));
```
MAGIC

damirs11 replied:
Сменил это... чудо на random.nextInt(1000)

dreambot92 commented on src/main/java/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,91 @@
+
+    public void printAll(){
+        Map<String, List<Document> > docGrouped =
+                setOfDocuments.stream().collect(Collectors.groupingBy(Document::getAuthor));
```
уверен, что стрим можно собрать сразу в treemap.. и вот эта штука будет не нужна docGrouped = new TreeMap<>(docGrouped);

dreambot92 commented on src/main/java/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,91 @@
+            System.out.print(k + ": \n");
+                v.forEach((doc) ->
+                        System.out.print(
+                                doc.getClass().getSimpleName() + " №" + doc.getIdDoc() + " от "
```
эту логику вывода можно засунуть в toString() для каждого класса документа

dreambot92 commented on src/main/java/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,91 @@
+                        System.out.print(
+                                doc.getClass().getSimpleName() + " №" + doc.getIdDoc() + " от "
+                                + doc.getRegDate() + ". " + doc.getName() + "\n")
+                );
```
А как тут реализована логика группировки по автору??

dreambot92 commented on src/main/java/com/company/Factory/IncomingFactory.java:
```diff
@@ -0,0 +1,56 @@
+    public Document generateData(Document doc) throws DocumentExistsException {
+
+        String[] names = {
+                "Смирнов Иванов",
```
эти имена есть в DocumentFactory.. бери их оттуда

damirs11 replied:
Создал отдельный класс для этого``

dreambot92 commented on src/main/java/com/company/Factory/IncomingFactory.java:
```diff
@@ -0,0 +1,56 @@
+public class IncomingFactory extends DocumentFactory {
+
+
+    Document create() throws DocumentExistsException {
```
по хорошему бы сделать интерфейс для фабрики с методом create()

damirs11 replied:
Сделал

dreambot92 commented on src/main/java/com/company/Factory/IncomingFactory.java:
```diff
@@ -0,0 +1,56 @@
+        };
+
+
+        GregorianCalendar gc = new GregorianCalendar();
```
знакомый код.. повторение не есть хорошо

damirs11 replied:
Исправил

dreambot92 commented on src/main/java/com/company/Models/Documents/Document.java:
```diff
@@ -0,0 +1,104 @@
+    private Date    regDate;
+    private String  author;
+
+    public Document() {
```
abstract и конструктор??

dreambot92 replied:
конструкторы в принципе допустимы в абстрактных классах при условии, что будут создаваться экземпляры его потомков.. но здесь это не предусмотрено

damirs11 replied:
Убрал

dreambot92 commented on src/main/java/com/company/Models/Documents/Document.java:
```diff
@@ -0,0 +1,104 @@
+    private int     idDoc;
+    private String  name;
+    private String  text;
+    private int     reg_idDoc;
```
прочитай про верблюжий стиль кода

damirs11 replied:
Исправил

dreambot92 commented on src/main/java/com/company/Models/Documents/Document.java:
```diff
@@ -0,0 +1,104 @@
+
+
+    public int compareTo(Document obj) {
+        if(this.reg_idDoc == obj.reg_idDoc)
```
используй compareTo
this.reg_idDoc.compareTo(obj.reg_idDoc);

dreambot92 commented on src/main/java/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,91 @@
+        setOfDocuments = new ArrayList<>();
+    }
+
+    public Document create(Class<? extends Document> docType) throws DocumentExistsException {
```
Еще бы не помешало логгировать ключевые события.. подключи логгер

stem25 commented on src/main/java/com/company/Factory/TaskFactory.java:
```diff
@@ -0,0 +1,56 @@
+    @Override
+    public Document generateData(Document doc) throws DocumentExistsException {
+
+        String[] names = {
```
Создай класс для получения персон(например PersonService) в нем добавь статическое поле List<String> persons
и не статичный метод получения списка персон List<String> getList(); и везде где тебе нужен будет список персон обращайся к этому сервису

damirs11 replied:
А будет ли хорошим способ вынести всю логику генерации рандомных значений для моей задачи в отдельный класс начиная от Персон и заканчивая генерацией даты ?

damirs11 replied:
Сделал

stem25 commented on src/main/java/com/company/Factory/TaskFactory.java:
```diff
@@ -0,0 +1,56 @@
+                "Петров Смирнов"
+        };
+
+        GregorianCalendar gc = new GregorianCalendar();
```
Вынеси генерацию даты в утилитный класс(например DateUtils)

damirs11 replied:
Исправил

stem25 commented on src/main/java/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,91 @@
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class DocumentFactory  {
```
сделай интерфейса для фабрики, а этот класс сделай абстрактным

damirs11 replied:
Исправил

damirs11 replied:
Сделал

stem25 commented on src/main/java/com/company/Interfaces/Storable.java:
```diff
@@ -0,0 +1,7 @@
+package com.company.Interfaces;
+
+public interface Storable {
+    int getIdDoc();
```
не надо писать(касается всех классов) idDoc и так понятно раз id находится в классе Document значит это идентификатор документа

damirs11 replied:
Исправил

stem25 commented on src/main/java/com/company/Models/Documents/Task.java:
```diff
@@ -0,0 +1,81 @@
+
+public class Task extends Document {
+
+    private Date    dateRealize;
```
Почитай code style, который Игорь скинул и во всем проекте поправь табуляцию и whitespace(idea: ctrl + alt + L)

stem25 commented on src/main/java/com/company/Main.java:
```diff
@@ -0,0 +1,13 @@
+public class Main {
+
+    public static void main(String[] args) {
+
```
Цель 1-2 задачи сделать исполняемый jar файл, смогу ли я собрать проект(mvn clean package) и выполнив java -jar "название приложение".jar получить требуемый результат ?

damirs11 replied:
Обновил pom.xml 
Теперь все собирается нормально

azamat2 commented on src/main/java/com/company/Exception/DocumentExistsException.java:
```diff
@@ -0,0 +1,11 @@
+
+public class DocumentExistsException extends Exception {
+    public DocumentExistsException() {
+        super("Документ с таким регистрационным номером уже существует");
```
можно сделать исключение чуть более информативным, добавив в него рег.номер

damirs11 replied:
Добавил

Commit:
Add ListWrappers for models from Staff

damirs11 committed:
.gitignore updated

damirs11 commented on .idea/encodings.xml:
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Исправил

damirs11 commented on pom.xml:
```diff
@@ -0,0 +1,52 @@
+
+    <dependencies>
+        <dependency>
+            <groupId>javax.xml.bind</groupId>
```
Разделил ветку на 2 задания
Как исправлю 1ое задание - удалю это ветку

damirs11 commented on pom.xml:
```diff
@@ -0,0 +1,52 @@
+        <dependency>
+            <groupId>javax.activation</groupId>
+            <artifactId>javax.activation-api</artifactId>
+            <version>1.2.0</version>
```
Исправил

damirs11 committed:
Исправил множество замечаний

Из основных
Сделал DocumentFactory абстрактным
Вынес генерацию рандомных данных/значений в отдельные классы
Поправил стилистику кода под требования

dreambot92 commented on src/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,72 @@
+    public static Document create(Class<? extends Document> docType) {
+
+        if(Objects.equals(docType.getName(), Outgoing.class.getName() )) {
+            return new OutgoingFactory().create();
```
Пусть create() просто создает экземпляр нужного класса, как и подобает фабрике.
А утилитки DataGeneratorUtils будут их заполнять

damirs11 replied:
Исправил

dreambot92 commented on src/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,72 @@
+        DateUtils dateUtils = new DateUtils();
+        DataGeneratorUtils dataGeneratorUtils = new DataGeneratorUtils();
+
+        doc.setIdDoc(random.nextInt(1000));
```
Прочитай про магические цифры. Их стоит вынести в константы

damirs11 replied:
Поправил

dreambot92 commented on src/com/company/Main.java:
```diff
@@ -0,0 +1,25 @@
+        //TODO подключить логгер
+        //TODO переписать toString() методы для моделей
+        //TODO проверить табуляцию
+        System.out.print(outgoing);
```
Давай сделаем что-нибудь по сложнее.. пусть будет цикл, который рандомно создает различные типы документов

damirs11 replied:
Добавил getRandomDocType() в DataGeneratorUtils

dreambot92 commented on src/com/company/Utils/DataGeneratorUtils.java:
```diff
@@ -0,0 +1,45 @@
+import java.util.List;
+import java.util.Random;
+
+public class DataGeneratorUtils {
```
Пусть этот класс занимается генерацией рандомных значений для документов. Подумай как это можно отрефакторить

damirs11 replied:
Вынес генерацию данных для разных документов в отдельные методы в классе DataGenerationUtils

dreambot92 commented on src/com/company/Utils/DataGeneratorUtils.java:
```diff
@@ -0,0 +1,45 @@
+    private Random random = new Random();
+
+    public DataGeneratorUtils() {
+    }
```
Для классов утилит конструкторы стоит сделать приватными

damirs11 replied:
Исправил

damirs11 replied:
Поправил

dreambot92 commented on src/com/company/Utils/DateUtils.java:
```diff
@@ -0,0 +1,31 @@
+    private Calendar calendar = Calendar.getInstance();
+    private Random random = new Random();
+
+    public Date takeRandomDate(int lowerBound, int upperBound){
```
здесь тоже все сделай статичным

damirs11 replied:
Поправил

dreambot92 commented on src/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,72 @@
+import java.util.Objects;
+import java.util.Random;
+
+public abstract class  DocumentFactory implements Factory {
```
javadoc-и

dreambot92 commented on src/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,72 @@
+
+    public Document generateData(Document doc) {
+
+        DateUtils dateUtils = new DateUtils();
```
подумай как перенести логику генерации в DataGeneratorUtils

damirs11 replied:
Перенес

damirs11 committed:
Вынес DateUtils в DataGeneratorUtils

damirs11 committed:
Add SLF4J logger, done

damirs11 committed:
Update pom.xml

dreambot92 commented on src/com/company/Factory/IncomingFactory.java:
```diff
@@ -0,0 +1,19 @@
+        Incoming doc = new Incoming(1, "", "", 1, new Date(), "");
+
+        DataGeneratorUtils.generateRandomDataForDocument(doc);
+        DataGeneratorUtils.generateRandomDataForIncoming(doc);
```
Два метода вызывать не нужно

damirs11 replied:
Исправил

dreambot92 commented on src/com/company/Models/Documents/Incoming.java:
```diff
@@ -0,0 +1,74 @@
+    }
+
+    public Incoming(Document otherDoc, String sender, String addressee, String outgoingNumber, Date outgoingDate) {
+        super(otherDoc);
```
Жесть!!))).. созданием инстанса у нас занимается фабрика и так, как дальше значения рандомизируются сами нам такое обилие конструкторов ни к чему.. пусть вызывается пустой конструктор

damirs11 replied:
Удалил

dreambot92 commented on src/com/company/Utils/DataGeneratorUtils.java:
```diff
@@ -0,0 +1,133 @@
+        int regId = random.nextInt(1000);
+
+        doc.setId(docId);
+        doc.setName("Название документа");
```
пусть логика будет построена таким образом, чтобы вызывался только этот метод для рандомизации

damirs11 replied:
Сделал методы перегруженными

dreambot92 commented on src/com/company/Utils/DateUtils.java:
```diff
@@ -0,0 +1,31 @@
+    public Date takeRandomDate(int lowerBound, int upperBound){
+
+        int year = random.nextInt(upperBound - lowerBound) + lowerBound;
+        int dayOfYear = random.nextInt(gc.getActualMaximum(Calendar.DAY_OF_YEAR)) + 1;
```
Как по мне наличия класса DataGeneratorUtils вполне достаточно.. пусть все будет там обрабатыватьяс

damirs11 replied:
Не правильно закоммитил из за чего получилась смесь из всего что у меня было
Сейчас поправлю

dreambot92 commented on src/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,29 @@
+        if(Objects.equals(docType.getName(), Task.class.getName() )) {
+            return new TaskFactory().create();
+        }
+
```
Вообще предлагаю метод DataGenerator-а засунуть сюда сразу после создания объекта

damirs11 replied:
Сделал

dreambot92 commented on src/com/company/Main.java:
```diff
@@ -0,0 +1,49 @@
+
+            for (Document document : documents) {
+                if (document.getRegId() == doc.getRegId()) {
+                    throw new DocumentExistsException(doc);
```
Неее.. пусть эта проверка будет при создании документа

damirs11 replied:
Засунул ее в Doc.Factory

damirs11 committed:
Delete useless code

damirs11 commented on src/com/company/Utils/DataGeneratorUtils.java:
```diff
@@ -0,0 +1,133 @@
+
+    public static Date addDays(Date date, int days){
+
+        Calendar calendar = Calendar.getInstance();
```
Исправил

damirs11 commented on pom.xml:
```diff
@@ -0,0 +1,56 @@
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.8.0</version>
```
Исправил

damirs11 commented on pom.xml:
```diff
@@ -0,0 +1,56 @@
+                <version>3.8.0</version>
+                <configuration>
+                    <source>8</source>
+                    <target>8</target>
```
Исправил

damirs11 commented on src/com/company/Utils/DataGeneratorUtils.java:
```diff
@@ -0,0 +1,133 @@
+        return deliveryType.get(random.nextInt(deliveryType.size()));
+    }
+
+    public static Class<? extends Document> getRandomDocType() {
```
Исправил

damirs11 commented on src/com/company/Utils/DataGeneratorUtils.java:
```diff
@@ -0,0 +1,133 @@
+            "И Соколов Попов"
+    );
+
+    private static List<String> deliveryType = Arrays.asList(
```
Исправил

damirs11 commented on src/com/company/Factory/IncomingFactory.java:
```diff
@@ -0,0 +1,19 @@
+public class IncomingFactory extends DocumentFactory {
+
+    public Document create() {
+        Incoming doc = new Incoming(1, "", "", 1, new Date(), "");
```
Исправил

damirs11 committed:
Update pom.xml
Changed List to Enum and add Enum Randomizer method in DataGeneratorClass
Changed addClass to Apache DateUtils.addDays
Overload generateRandomDataForDocument
Delete useless constructors in models
Add checkId which throw exception if id both documents Are equal

damirs11 committed:
Fix checkid between same objects
Replace standard if-statement to compareTo in DocumentModel

damirs11 committed:
minor fixes

damirs11 committed:
update takeRandomDate

dreambot92 commented on src/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,65 @@
+    private static void checkId(Document doc) throws DocumentExistsException {
+        int count = 0;
+        for(Document document: Document.allDocuments) {
+            if(document.getRegId().equals(doc.getRegId()) && (count == 2)){
```
Что это за count == 2??

dreambot92 replied:
и что это за count вообще?

dreambot92 replied:
Вообще, может имеет смысл генерацию регистрационного номера и его дальнейшей проверки вынести именно в этот метод

damirs11 replied:
Ошибся случайно глубокой ночью хд 
А так "++count > 2" должен быть 
т.е. если попался документ с таким же id не считая себя самого, то выкинуть искл.

damirs11 replied:
Исправил

dreambot92 commented on src/com/company/Models/Documents/Document.java:
```diff
@@ -0,0 +1,100 @@
+
+    public static final List<Document> allDocuments;
+    static   {
+        allDocuments = new ArrayList<>();
```
final List? а оно так работает??
Так или иначе, лучше эту логику куда-нибудь вынести

dreambot92 commented on src/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,65 @@
+     * @return  Document with generated data
+     * @throws DocumentExistsException if idReg both documents are identical
+     */
+    public static Document create(Class<? extends Document> docType) throws DocumentExistsException {
```
используй docTypes

damirs11 replied:
Заменил

damirs11 commented on src/com/company/Utils/DataGeneratorUtils.java:
```diff
@@ -0,0 +1,145 @@
+    }
+
+    private enum persons {
+        PERSONS1 ("А Иванов Кузнецов"),
```
Исправил

damirs11 commented on src/com/company/Utils/DataGeneratorUtils.java:
```diff
@@ -0,0 +1,145 @@
+        return randomEnum(docTypes.class).getType();
+    }
+    public static Date takeRandomDate(){
+        return new Date(System.currentTimeMillis() - ThreadLocalRandom.current().nextLong(1000*60*60*24*1000));
```
Заменил

damirs11 commented on src/com/company/Utils/DataGeneratorUtils.java:
```diff
@@ -0,0 +1,145 @@
+    public static Document generateRandomDataForDocument(Document doc){
+
+        Integer docId = random.nextInt(1000);
+        Integer regId = random.nextInt(1000);
```
заменил

damirs11 commented on src/com/company/Factory/DocumentFactory.java:
```diff
@@ -0,0 +1,65 @@
+            DataGeneratorUtils.generateRandomDataForDocument(doc);
+
+            checkId(doc);
+
```
Вынес

damirs11 commented on src/com/company/Utils/DataGeneratorUtils.java:
```diff
@@ -0,0 +1,145 @@
+    }
+
+    private enum deliveryType {
+        TYPE1 ("Способ доставки 1"),
```
Вынес

damirs11 committed:
Refactor struct of proj

damirs11 committed:
minor fixes add docs to parsers

damirs11 committed:
minor fixes

damirs11 committed:
add DocumentsStorage and replace
DocumentFactory.checkId() this DocumentsStorage.add(doc)

damirs11 committed:
Delete comments

dreambot92 commented on src/main/java/com/company/Main.java:
```diff
@@ -0,0 +1,42 @@
+        }
+
+        //Сортировка List в TreeMap
+        TreeMap<String, List<Document>> docsGrouped = documents.stream()
```
`Map<String, List<Document>>` .. а может `Map<String, Collection<Document>>`
Если задумаем использовать не список, а допустим множество

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/enumeration/docTypes.java:
```diff
@@ -0,0 +1,22 @@
+
+public enum docTypes {
+    OUTGOING (Outgoing.class),
+    INCOMING (Incoming.class),
```
реализовать без классов

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/factory/DocumentFactory.java:
```diff
@@ -0,0 +1,53 @@
+public abstract class DocumentFactory implements Factory {
+
+    private DocumentFactory() {
+        throw new IllegalStateException();
```
удалить

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/factory/DocumentFactory.java:
```diff
@@ -0,0 +1,53 @@
+
+        if(Objects.equals(docType.getType(), Outgoing.class )) {
+            doc = OutgoingFactory.create();
+            DataGeneratorUtils.generateRandomDataForDocument( (Outgoing) doc);
```
вынести generateRandomDataForDocument за условия

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/factory/IncomingFactory.java:
```diff
@@ -0,0 +1,14 @@
+class IncomingFactory {
+
+    private IncomingFactory() {
+        throw new IllegalStateException();
```
удалить

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/models/documents/Document.java:
```diff
@@ -0,0 +1,110 @@
+    private String author;
+
+    public Document() {
+    }
```
удалить

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/storage/DocumentsStorage.java:
```diff
@@ -0,0 +1,31 @@
+    }
+
+    private static List<Document> documentList = new ArrayList<>();
+
```
хранить регистрационные номера вместо документов

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/utils/DataGeneratorUtils.java:
```diff
@@ -0,0 +1,103 @@
+        doc.setId(random.nextInt(1000));
+        doc.setName("Название документа");
+        doc.setText("Текс документа");
+        doc.setAuthor(takeRandomPerson());
```
рефакторинг

damirs11 replied:
+

damirs11 committed:
исправил мелкие замечания, но еще не все

damirs11 committed:
Начала работу над аннотациями

damirs11 committed:
Done

dreambot92 commented on src/main/java/com/company/Main.java:
```diff
@@ -0,0 +1,42 @@
+
+public class Main {
+
+    public static void main(String[] args) throws DocumentExistsException, IllegalAccessException {
```
Давай здесь обрабатывать исключения через try catch..  в этом методе пробрасывать дальше смысла нет

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/Main.java:
```diff
@@ -0,0 +1,42 @@
+
+        for(int i = 0; i < 10; i++){
+            Document doc = DocumentFactory.create(DataGeneratorUtils.getRandomDocType());
+            documents.add(doc);
```
Давай сократим чутка
documents.add(DocumentFactory.create(DataGeneratorUtils.getRandomDocType()))

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/models/documents/Incoming.java:
```diff
@@ -0,0 +1,77 @@
+    @RandomValue(value = randomTypes.DATE)
+    private Date outgoingDate;
+
+    public Incoming() {
```
пустой конструктор

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/models/documents/Outgoing.java:
```diff
@@ -0,0 +1,53 @@
+    private String deliveryType;
+
+    public Outgoing() {
+    }
```
тоже удалить

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/models/documents/Task.java:
```diff
@@ -0,0 +1,87 @@
+    private String controller;
+
+    public Task() {
+    }
```
удалить

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/storage/DocumentsStorage.java:
```diff
@@ -0,0 +1,27 @@
+    private DocumentsStorage() {
+    }
+
+    private static Collection<Integer> documentList = new ArrayList<>();
```
переименовать бы.. у нас здесь номера хранятся, а не документы

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/utils/RandomValueProcessor.java:
```diff
@@ -0,0 +1,56 @@
+        fields.addAll(Arrays.asList(clazz.getDeclaredFields()));
+        fields.addAll(Arrays.asList(superClazz.getDeclaredFields()));
+
+        for(Field field: fields) {
```
можно сократить так ^^
```
for(Field field: fields) if (field.isAnnotationPresent(RandomValue.class)) {
...
}
```

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/utils/RandomValueProcessor.java:
```diff
@@ -0,0 +1,56 @@
+            }
+
+            randomTypes type = field.getAnnotation(RandomValue.class).value();
+            int upperBound = field.getAnnotation(RandomValue.class).upperBound();
```
мне кажется эта переменная ни к чему

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/utils/RandomValueProcessor.java:
```diff
@@ -0,0 +1,56 @@
+
+        Collection<Field> fields = new ArrayList<>();
+        fields.addAll(Arrays.asList(clazz.getDeclaredFields()));
+        fields.addAll(Arrays.asList(superClazz.getDeclaredFields()));
```
Интересно.. а clazz.getDeclaredFields() не содержит уже в себе родительские поля??
Если не содержит используй org.apache.hadoop.util.ReflectionUtils

dreambot92 commented on src/main/java/com/company/utils/RandomValueProcessor.java:
```diff
@@ -0,0 +1,56 @@
+    private static Random random = new Random();
+
+    public static void proccess(Object obj) throws IllegalAccessException {
+        Class clazz = obj.getClass();
```
этот метод можно и в DataGeneratorUtils засунуть..

damirs11 replied:
+

damirs11 committed:
Add javadocs
Minor fixes

dreambot92 commented on src/main/java/com/company/utils/DataGeneratorUtils.java:
```diff
@@ -0,0 +1,124 @@
+        Class clazz = obj.getClass();
+        Collection<Field> fields = ReflectionUtils.getDeclaredFieldsIncludingInherited(clazz);
+
+        for (Field field : fields)
```
не будем создавать лишние переменные
for (Field field : ReflectionUtils.getDeclaredFieldsIncludingInherited(obj.getClass()))

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/factory/Factory.java:
```diff
@@ -0,0 +1,9 @@
+import com.company.exception.DocumentExistsException;
+import com.company.models.documents.Document;
+
+public interface Factory {
```
тут еще

dreambot92 commented on src/main/java/com/company/factory/IncomingFactory.java:
```diff
@@ -0,0 +1,9 @@
+
+import com.company.models.documents.Incoming;
+
+class IncomingFactory {
```
и тут

dreambot92 commented on src/main/java/com/company/factory/OutgoingFactory.java:
```diff
@@ -0,0 +1,9 @@
+
+import com.company.models.documents.Outgoing;
+
+class OutgoingFactory {
```
и тут

dreambot92 commented on src/main/java/com/company/factory/TaskFactory.java:
```diff
@@ -0,0 +1,9 @@
+
+import com.company.models.documents.Task;
+
+class TaskFactory {
```
и тут

dreambot92 commented on src/main/java/com/company/models/documents/Document.java:
```diff
@@ -0,0 +1,106 @@
+import java.util.Date;
+import java.util.Objects;
+
+public abstract class Document implements Comparable<Document>, Storable {
```
тут тоже

dreambot92 commented on src/main/java/com/company/models/documents/Incoming.java:
```diff
@@ -0,0 +1,74 @@
+import java.util.Date;
+import java.util.Objects;
+
+public class Incoming extends Document {
```
тут

dreambot92 commented on src/main/java/com/company/models/documents/Outgoing.java:
```diff
@@ -0,0 +1,50 @@
+
+import java.util.Objects;
+
+public class Outgoing extends Document {
```
тут

dreambot92 commented on src/main/java/com/company/models/documents/Task.java:
```diff
@@ -0,0 +1,84 @@
+import java.util.Date;
+import java.util.Objects;
+
+public class Task extends Document {
```
тут

dreambot92 commented on src/main/java/com/company/storage/IdDocumentsStorage.java:
```diff
@@ -0,0 +1,37 @@
+    private static Collection<Integer> idDocumentsList = new ArrayList<>();
+
+    /**
+     *
```
на геттеры, сеттеры javadoc-и не нужны

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/models/documents/Document.java:
```diff
@@ -0,0 +1,106 @@
+    private String text;
+    @RandomValue(value = randomTypes.INTEGER)
+    private Integer regId;
+    @RandomValue(value = randomTypes.DATE)
```
в аннотации задано только одно значение.. можно сократить
@RandomValue(randomTypes.DATE)

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/utils/DataGeneratorUtils.java:
```diff
@@ -0,0 +1,124 @@
+     * @param obj the target object
+     * @throws IllegalAccessException the illegal access exception
+     */
+    public static void process(Object obj) throws IllegalAccessException {
```
может лучше обозвать generate().. и на входе мы ждем объект класса Document, это тоже отрази

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/utils/DataGeneratorUtils.java:
```diff
@@ -0,0 +1,124 @@
+     * Random value from enum
+     *
+     * @param <T>   the type of enum
+     * @param clazz the clazz of enum
```
у тебя тут только clazz в параметрах

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/enumeration/deliveryType.java:
```diff
@@ -0,0 +1,21 @@
+/**
+ * The enum Delivery type.
+ */
+public enum deliveryType {
```
только щас обратил внимание.. с заглавной буквы
Все остальные название enum-ов тоже исправь

damirs11 replied:
+

dreambot92 commented on src/main/java/com/company/utils/DataGeneratorUtils.java:
```diff
@@ -0,0 +1,124 @@
+                Object value = null;
+                switch (field.getAnnotation(RandomValue.class).value()) {
+                    case INTEGER:
+                        value = random.nextInt(1000);
```
магические числа переведи в константы

damirs11 committed:
Add more javadocs

damirs11 committed:
Add more javadocs

damirs11 committed:
Minor fix
